e2e/test/AzureSecurityCenterForIoTSecurityMessageE2ETests.cs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Diagnostics.Tracing;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class AzureSecurityCenterForIoTSecurityMessageE2ETests : IDisposable
+    {
+        private readonly string _devicePrefix = $"E2E_{nameof(AzureSecurityCenterForIoTSecurityMessageE2ETests)}_";
+        private readonly string _modulePrefix = $"E2E_{nameof(AzureSecurityCenterForIoTSecurityMessageE2ETests)}_";
+        private static TestLogging _log = TestLogging.GetInstance();
+
+        private readonly ConsoleEventListener _listener;
+        private readonly AzureSecurityCenterForIoTLogAnalyticsClient _logAnalyticsClient;
+
+        public AzureSecurityCenterForIoTSecurityMessageE2ETests()
+        {
+            _listener = TestConfig.StartEventListener();
+            _logAnalyticsClient = AzureSecurityCenterForIoTLogAnalyticsClient.CreateClient();
+        }
+
+        [TestMethod]
+        public async Task SecurityMessage_DeviceSendSingleMessage_Amqp()
+        {
+            await TestSecurityMessage(Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
```

Comment:
Nit: for all these we can definitely avoid creating yet another state machine for a single call which comes with certain overheads and makes the stack harder in case of bugs:

```
public Task Security...() // no async
{
   return TestSecurityMessage(...); //no ConfigureAwait
}

Reply:
I tried to do this, but then the e2e test won't run if they're not async Task functions.

---
In reply to: [301725222](https://github.com/Azure/azure-iot-sdk-csharp/pull/999#discussion_r301725222) [](ancestors = 301725222)